### PR TITLE
Propagate result from http response slot

### PR DESCRIPTION
**Pull request by lbudai**

### CODE REVIEW

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by lbudai** [comment]
@bazsi : Yes, it could be possible - but my time capacity is very limited. When I opened the PR I just wanted to address one specific issue (immediate retry when an auth header successfully renewed a auth token when received HTTP 401). 
So my primary goal with these PRs is to eliminate the http-auth-header-context. As one of our internal module implemented as a http-auth-header, I also had to re-implement that module and solve all the related issues. One of them was how to react HTTP 401.

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Comment by kira-syslogng** [comment]
Build SUCCESS

File: modules/http/http-grammar.ym [hunk 1]
```diff
@@ -146,7 +145,6 @@ http_option
     | KW_PASSWORD   '(' string ')'            { http_dd_set_password(last_driver, $3); free($3); }
     | KW_USER_AGENT '(' string ')'            { http_dd_set_user_agent(last_driver, $3); free($3); }
     | KW_HEADERS    '(' string_list ')'       { http_dd_set_headers(last_driver, $3); g_list_free_full($3, free); }
-    | KW_AUTH_HEADER '(' http_auth_header_plugin ')' { http_dd_set_auth_header(last_driver, $3); }
```
[the remark this answers is not in the document]
Comment (lbudai) [reply modules/http/http-grammar.ym:149]: @Kokan : thanks!
fixed (squashed into the first commit).

**Comment by kira-syslogng** [comment]
Build SUCCESS

**Pull request merged**